restapi/src/main/java/io/stargate/web/docsapi/resources/CollectionsResource.java

```diff
@@ -117,34 +132,33 @@ public Response createCollection(
               value = "JSON with the name of the collection",
               required = true,
               example = "{\"name\": \"example\"}")
-          String payload,
+          @NotNull(message = "payload not provided")
+          @Valid
+          CreateCollectionDto body,
       @Context HttpServletRequest request) {
     return RequestHandler.handle(
         () -> {
           Map<String, String> allHeaders = getAllHeaders(request);
           DocumentDB docDB = db.getDocDataStoreForToken(token, allHeaders);
-          DocCollection info = mapper.readValue(payload, DocCollection.class);
-          if (info.getName() == null) {
-            throw new IllegalArgumentException("`name` is required to create a collection");
-          }
+
           db.getAuthorizationService()
               .authorizeSchemaWrite(
                   docDB.getAuthenticationSubject(),
                   namespace,
-                  info.getName(),
+                  body.getName(),
                   Scope.CREATE,
                   SourceAPI.REST,
                   ResourceKind.TABLE);
 
-          boolean res = collectionService.createCollection(namespace, info.getName(), docDB);
+          boolean res = collectionService.createCollection(namespace, body.getName(), docDB);
           if (res) {
             return Response.status(Response.Status.CREATED).build();
           } else {
             return Response.status(Response.Status.CONFLICT)
                 .entity(
                     new Error(
                         String.format(
-                            "Create failed: collection %s already exists.", info.getName()),
+                            "Create failed: collection %s already exists.", body.getName()),
                         Response.Status.CONFLICT.getStatusCode()))
```

Comment:
Is there a way to avoid duplicate references to `Response.Status.CONFLICT` in this fragment?

Reply:
Sure, but let's keep this for now, as we would further address this once we refactor to Rx and actually move everything away from this class to the service I guess, ok? Then it's gonna be:

```
collectionService.something()
   .map(result -> Response...)
   .onErrorReturn(ErrorCodeRuntimeException.class, e -> e.toResponse())
   .onErrorReturn(Throwable.class, e -> ErrorCode.INTERAL_ERROR.toResponse(e.getMessage())
```

or smth in this direction..

Reply:
sgmt